src/Surfnet/Stepup/Identity/Identity.php

```diff
@@ -495,6 +498,51 @@ public function vetSecondFactor(
         );
     }
 
+    /**
+     * Self vetting, is when the user uses it's own token to vet another.
+     *
+     * Here the new token should have a lower or equal LoA to that of the one in posession of the user
+     */
+    public function selfVetSecondFactor(
+        SecondFactorId $authoringSecondFactorId,
+        string $registrationCode,
+        SecondFactorIdentifier $secondFactorIdentifier,
+        SecondFactorTypeService $secondFactorTypeService
+    ): void {
+        $this->assertNotForgotten();
+        $registeringSecondFactor = null;
+        foreach ($this->verifiedSecondFactors as $secondFactor) {
+            /** @var VerifiedSecondFactor $secondFactor */
+            if ($secondFactor->hasRegistrationCodeAndIdentifier($registrationCode, $secondFactorIdentifier)) {
+                $registeringSecondFactor = $secondFactor;
+            }
+        }
+
+        if ($registeringSecondFactor === null) {
+            throw new DomainException(
+                sprintf(
+                    'Registrant second factor of type %s with ID %s does not exist',
+                    get_class($secondFactorIdentifier),
+                    $secondFactorIdentifier->getValue()
+                )
+            );
+        }
+
+        if (!$registeringSecondFactor->hasRegistrationCodeAndIdentifier($registrationCode, $secondFactorIdentifier)) {
+            throw new DomainException('The verified second factors registration code or identifier do not match.');
+        }
+        $authoringSecondFactor = $this->getVettedSecondFactor($authoringSecondFactorId);
+        if ($authoringSecondFactor === null) {
+            throw new DomainException(
+                sprintf('Authorizing second factor with ID %s does not exist', $authoringSecondFactorId)
+            );
+        }
+        if ($authoringSecondFactor->hasEqualOrHigherLoaComparedTo($registeringSecondFactor, $secondFactorTypeService)) {
+            throw new DomainException("The second factor to be vetted has a higher LoA then the Token used for proving posession");
+        }
+        $registeringSecondFactor->vet(DocumentNumber::unknown(), true);
```

Comment:
Could we refactor the use of DocumentNumber so we could use it for RV and SV to show the type of vetting?

Reply:
Yes, ended up adding a vetting type interface which can be implemented for your preferred vetting type. I added the on premise and self vet varieties. The first has the document number, and the self vet option tracks the token loa the vet action was self vetted with. You should be able to create a similar remote vet option